src/api/app/jobs/send_event_emails.rb

```diff
@@ -33,4 +40,13 @@ def create_rss_notifications(event)
       )
     end
   end
+
+  def needs_logging?(event)
+    !event.project_logged && (event.class < Event::Project || event.class < Event::Package)
+  end
+
+  def create_project_log_entry(event)
```

Comment:
This could also be a the Event class... something like:
```ruby
 def create_log_entry
  unless logged?
     ProjectLogEntry.create_from(self)
     update_attributes(project_logged: true)
  end
end
```